source_code/main_mcu/src/COMMS/comms_hid_defines.h

```diff
@@ -125,6 +126,13 @@ typedef struct
     cust_char_t concatenated_strings[0];
 } hid_message_store_cred_t;
 
+typedef struct
+{
+    cust_char_t service_name[SERVICE_NAME_MAX_LEN];
```

Comment:
that's an interesting way to do it... requiring moolticute to pad the fields though... you didn't feel like doing like the other message? :p

Reply:
So the way the other message is doing it seems very, very dangerous, prone to bugs/error (read: security vulnerability) to me :-). I have always created protocols the way above with just fixed fields (KISS). Don't need tons of validation code and much less chance of bugs.

What is the reason for the packing format? To save some bytes in transfer?

Reply:
indeed, useful when doing it over bluetooth!
however in your case we may keep it this way